### PR TITLE
Clarify user migration in 1.x to 2.0 automated upgrade process

### LINKED ISSUE #2336: Description of InfluxDB user migration omits information
The "User migration" section of the 2.x -> 1.x upgrade docs say:

> influxd upgrade migrates existing 1.x users and their permissions. However, it does not migrate administrative users.
> 
> To review users with admin permissions, in the InfluxDB 1.x CLI, run show users. Any users labeled “admin” will not be migrated.

This is all true, but it isn't the full story. The upgrade process _also_ drops any non-admin users that don't have any privileges. This has caused confusion for at least 1 user.

##### Relevant URLs
- Original report: influxdata/influxdata-docker#472
- Relevant docs: https://docs.influxdata.com/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade/#user-migration

### PR DESCRIPTION
**Pull request by sanderson**
Closes #2336 

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md [hunk 1]
```diff
@@ -101,10 +101,14 @@ You can continue to use Kapacitor with InfluxDB OSS 2.0 under the following scen
 ### User migration
 
 `influxd upgrade` migrates existing 1.x users and their permissions.
-However, it *does not migrate administrative users*.
+However, it **does not** migrate the following users:
 
-To review users with admin permissions, in the InfluxDB 1.x CLI, run `show users`.
-Any users labeled "admin" *will not* be migrated.
+- administrative (admin) users
+- non-admin users without privileges
```
Comment (kelseiv) [inline content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:107]: Users without admin permissions? 
Are admin users and non-admin users distinct in ways other than who has admin permissions? If yes, do we define this distinction?

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md [hunk 2]
```diff
@@ -101,10 +101,14 @@ You can continue to use Kapacitor with InfluxDB OSS 2.0 under the following scen
 ### User migration
 
 `influxd upgrade` migrates existing 1.x users and their permissions.
-However, it *does not migrate administrative users*.
+However, it **does not** migrate the following users:
```
Comment (kelseiv) [inline content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:104]: `influxd upgrade` migrates existing 1.x users and their permissions **except** the following users:

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md [hunk 3]
```diff
@@ -101,10 +101,14 @@ You can continue to use Kapacitor with InfluxDB OSS 2.0 under the following scen
...
 
-To review users with admin permissions, in the InfluxDB 1.x CLI, run `show users`.
-Any users labeled "admin" *will not* be migrated.
+- administrative (admin) users
+- non-admin users without privileges
+
+To review users with admin permissions, run
+[`SHOW USERS`](/{{< latest "influxdb" "v1" >}}/administration/authentication_and_authorization/#show-all-existing-users-and-their-admin-status)
+against your InfluxDB 1.x instance.
```
Comment (kelseiv) [inline content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:111]: After finding users with admin permissions, manually migrate...?

**Review by kelseiv — APPROVED** [review]

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md (hunk 1 above)
Comment (danxmoran) [reply content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:107]: The 1.x docs make a clear divide between admin and non-admin: https://docs.influxdata.com/influxdb/v1.8/administration/authentication_and_authorization/#user-types-and-privileges

My brain likes something like this better:
```suggestion
- administrative (admin) users
- non-admin users that have not been granted any privileges
```
So if people want to make sure all non-admins are migrated, they see there's an action they can take (granting privileges) to make it happen

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md (hunk 3 above)
Comment (danxmoran) [reply content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:111]: I'm honestly not sure if there's a parallel 2.x entity that users could manually migrate to, other than the initial super-user token created by `influxd upgrade` or `influx setup`

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md [hunk 4]
```diff
@@ -101,10 +101,14 @@ You can continue to use Kapacitor with InfluxDB OSS 2.0 under the following scen
...
```
Comment (danxmoran) [inline content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:112]: Should we also add a paragraph for "To review non-admin users with no permissions, ..."?

Comment (sanderson) [reply content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:124]: Good call, added.

**Review by danxmoran — APPROVED** [review]

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md [hunk 5]
```diff
@@ -100,18 +100,41 @@ You can continue to use Kapacitor with InfluxDB OSS 2.0 under the following scen
...
+**create a new read-only user before upgrading**:
+
+##### Create a read-only 1.x user
+```sh
+> CREATE USER <username> WITH PASSWORD '<password>'
+> GRANT READ ON <database> TO "<username>"
+```
+
+InfluxDB 2.0 only grants admin privileges to the primary user created in the
+InfluxDB 2.0 setup process which runs at the end of the upgrade process.
```
Comment (kelseiv) [inline content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:135]:
```suggestion
InfluxDB 2.0 only grants admin privileges to the primary user set up during the InfluxDB 2.0 upgrade.
```

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md [hunk 6]
```diff
@@ -100,18 +100,41 @@ You can continue to use Kapacitor with InfluxDB OSS 2.0 under the following scen
...
+This provides you with the opportunity to reassess who to grant admin-level
+access to in your InfluxDB 2.0 setup.
```
Comment (kelseiv) [inline content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:137]:
```suggestion
This provides you with the opportunity to reassess who to grant admin permissions when setting up InfluxDB 2.0.
```

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md [hunk 7]
```diff
@@ -100,18 +100,41 @@ You can continue to use Kapacitor with InfluxDB OSS 2.0 under the following scen
...
+  that have not been granted any privileges
+
+{{< expand-wrapper >}}
+{{% expand "Review 1.x user privileges" %}}
+**To review 1.x users with admin privileges**, run the following against your InfluxDB 1.x instance:
+
+```sql
+SHOW USERS
+```
+
```
Comment (kelseiv) [inline content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:116]: thoughts on adding: Any users labeled “admin” will not be migrated.

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md (hunk 6 above)
Comment (kelseiv) [reply content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:137]: Ideally, we'd include link to setting up InfluxDB, but given this doc isn't entirely task-based, it interferes with the workflow as is. For future thought...

**Review by kelseiv — APPROVED** [review]

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md [hunk 8]
```diff
@@ -100,18 +100,41 @@ You can continue to use Kapacitor with InfluxDB OSS 2.0 under the following scen
...
+**To review the specific privileges granted to each 1.x user**, run the following for each user in your InfluxDB 1.x instance:
+
+```sql
+SHOW GRANTS FOR "<username>"
+```
+{{% /expand %}}
+{{< /expand-wrapper >}}
```
Comment (kelseiv) [inline content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:123]: thoughts on adding: If no grants appear, the user labeled “admin” will not be migrated.

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md (hunk 7 above)
Comment (sanderson) [reply content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:116]: 👍🏻

File: content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md (hunk 8 above)
Comment (sanderson) [reply content/influxdb/v2.0/upgrade/v1-to-v2/automatic-upgrade.md:123]: 👍🏻

**Pull request merged**